packages/core/src/tracing/instrumentation/cloud/aws-sdk/v2/sqs.js

```diff
@@ -355,53 +356,103 @@ function instrumentedReceiveMessage(ctx, originalReceiveMessage, originalArgs) {
   });
 }
 
-function readAttribCaseInsensitive(attributes, key1, key2) {
-  return (
-    tracingUtil.readAttribCaseInsensitive(attributes, key1) || tracingUtil.readAttribCaseInsensitive(attributes, key2)
-  );
-}
-
 /**
- * Flattens the AWS SQS MessageAttribute format into a basic object structure.
+ * Reads all trace context relevant message attributes from an incoming message and provides them in a normalized format
+ * for later processing.
  */
-function convertAttributesFromSQS(sqsAttributes) {
-  const attributes = {};
+function readTracingAttributes(sqsAttributes) {
```

Comment:
This works a little different now than before. Instead of copying/flattening all incoming attributes and using that flattened copy, we now only read the three attributes we care about and put them into a normalized structure of the shape: `{traceId, parentId, level}`.  

All code that is executed after that then only needs to work with that uniform structure and does not need to care about legacy header names etc.